moveit_core/collision_detection/include/moveit/collision_detection/collision_common.h

```diff
@@ -217,6 +220,139 @@ struct CollisionRequest
   /** \brief Flag indicating whether information about detected collisions should be reported */
   bool verbose;
 };
+
+struct DistanceRequest
+{
+  DistanceRequest()
+    : enable_nearest_points(false)
+    , enable_signed_distance(false)
+    , global(true)
+    , active_components_only(nullptr)
+    , acm(nullptr)
+    , distance_threshold(std::numeric_limits<double>::max())
+    , verbose(false)
+  {
+  }
+
+  virtual ~DistanceRequest()
+  {
+  }
+
+  /// Compute \e active_components_only_ based on \e req_
+  void enableGroup(const robot_model::RobotModelConstPtr& kmodel)
+  {
+    if (kmodel->hasJointModelGroup(group_name))
+      active_components_only = &kmodel->getJointModelGroup(group_name)->getUpdatedLinkModelsWithGeometrySet();
+    else
+      active_components_only = nullptr;
+  }
+
+  /// Indicate if nearest point information should be calculated
+  bool enable_nearest_points;
+
+  /// Indicate if a signed distance should be calculated in a collision.
+  bool enable_signed_distance;
+
+  /// Indicate if the global minimum should only be found. If this is true
+  /// it will only try to find the global minimum distance and not store information
+  /// on a link by link basis. If this is set to false it will store distance information
+  /// for every link in the active_components_only list.
+  bool global;
+
+  std::string group_name;
+
+  /// The set of active components to check
+  const std::set<const robot_model::LinkModel*>* active_components_only;
+
+  /// The allowed collision matrix used to filter checks
+  const AllowedCollisionMatrix* acm;
+
+  /// A distance threshold to reduce number of queries
+  double distance_threshold;
+
+  /// Log debug information
+  bool verbose;
+
+  /// Indicate if gradient should be calculated between each object. This the vector defined by the line connecting the
+  /// closest points on the two objects.
+  bool gradient;
+};
+
+struct DistanceResultsData
+{
+  DistanceResultsData()
+  {
+    clear();
+  }
+
+  /// The distance between two objects. If two objects are in collision, distance <= 0.
+  double distance;
+
+  /// The nearest points
+  Eigen::Vector3d nearest_points[2];
+
+  /// The object link names
+  std::string link_names[2];
+
+  /// A normalized vector pointing from link_names[0] to link_names[1].
+  Eigen::Vector3d normal;
+
+  /// Indicates if nearest points were found.
```

Comment:
That is, whether `nearest_points[]` is valid?
What are reasons, that nearest point are not found?
Please augment the documentation here.

Reply:
I wrote most of this a few years ago so not sure what my intent was at the time, but after reviewing it it should probably be removed.